### PR TITLE
Feature/uvc support

### PR DESCRIPTION
**Pull request by thaemmerle**
What is missing:
- attributes for specific format descriptors (uncompressed and mjpeg)
- all "get-functionalities"

Please let me know if you have any comments or suggestions for improvement!

### CODE REVIEW

**Comment by pszewczyk** [comment]
If you mean the complex, nested structure you created for handling uvc attributes, it's fine by me. It matches the actual configuration tree which should be familiar to potential users.

Of course, handling this in code attribute-by-attribute could be painful, but that's what import/export functionality is for. I also imagine the most convenient way to use this would be to create the gadget file with all parameters for uvc and load it by gt or from some piece of code. It would also be a 5-levels nested structure in config files, but I think libconfig should handle that pretty well. 
As for gadget-tool itself, it would be nice to add support for uvc function. But the most important functionality, that is import / export of whole gadgets is already there, so don't treat it as a high priority.

What I'm not convinced about is the 'path' parameter in setters/getters, e.g.:
`int usbg_f_uvc_set_frame(const char *path, const char *name, const char *format_desc, const char *spec_ format_desc, const struct usbg_f_uvc_frame_descriptor *frame_descriptor)`
Why not use the the function as a parameter, like in other function interfaces?

**Comment by thaemmerle** [comment]
Sry, for the late response:

Regarding the 'path' parameter:
I think this is consistent with the other functions of the lib. 
Take for example a decimal setting:
The ".set" function is set to `usbg_set_dec`.  `usbg_set_dec` also takes a parameter `const char *path`.
The only difference is that it makes no sense to move `usbg_f_uvc_set_format_descs` to `usbg_internal.h` because it is only used by UVC and settings in sub directories are set.

**Comment by thaemmerle** [comment]
I've now added get functionalities and changed and included the proposals from @pszewczyk 

What do you mean?

**Comment by kopasiak** [comment]
@thaemmerle 
Yes sure. It's not lack of interest just lack of time:(

File: src/function/uvc.c [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+} uvc_attr[USBG_F_UVC_ATTR_MAX] = {
+	[USBG_F_UVC_STREAMING_INTERVAL] = UVC_DEC_ATTR(streaming_interval),
+	[USBG_F_UVC_STREAMING_MAX_BURST] =		UVC_DEC_ATTR(streaming_maxburst),
+	[USBG_F_UVC_STREAMING_MAX_PACKET] = UVC_DEC_ATTR(streaming_maxpacket),
+	[USBG_F_UVC_FORMAT_DESCS] =		UVC_FORMAT_DESC_ATTR(format_descriptors),
+};
+
+static struct
+{
+	const char *name;
```
Comment (kopasiak) [inline src/function/uvc.c]: did I missed sth or it's not possible to get bInterfaceNumber from control dir?

Comment (kopasiak) [reply src/function/uvc.c]: What with terminal* attributes?

**Comment by thaemmerle** [comment]
I've tried to fit my code to the proposed amendments, what resulted in a huge change.

@pszewczyk and @kopasiak:
Please take a look at feature/uvc_support_v2 and let me know
what you think about the basic architecture so that I can go on with
the implementation.

**Pull request closed without merging**